### PR TITLE
BFD-2455: Unit tests for the contained resource for organization

### PR DESCRIPTION
**Pull request by cbrunefearless**
**JIRA Ticket:**
[BFD-2455](https://jira.cms.gov/browse/BFD-2455)

**User Story or Bug Summary:**

 

User Story:
These functions do not currently have any unit tests for the contained resource for organization.

CARIN Confluence Page - https://confluence.cms.gov/display/BFD/CARIN+Conformance+-+Eob

---

### What Does This PR Do?

Adds more hardening to our system by adding unit tests for the contained organization.

### What Should Reviewers Watch For?

If you're reviewing this PR, please check for these things in particular:

* Verify all PR security questions and checklists have been completed and addressed.

### What Security Implications Does This PR Have?

Submitters should complete the following questionnaire:

* If the answer to any of the questions below is **Yes**, then **you must** supply a link to the associated Security Impact Assessment (SIA), security checklist, or other similar document in Confluence here: **N/A**

    * Does this PR add any new software dependencies? 
      * [ ] Yes
      * [X] No
    * Does this PR modify or invalidate any of our security controls?
      * [ ] Yes
      * [X] No
    * Does this PR store or transmit data that was not stored or transmitted before?
      * [ ] Yes
      * [X] No

* If the answer to any of the questions below is **Yes**, then please add @StewGoin as a reviewer, and note that this PR **should not be merged** unless/until he also approves it.
    * Do you think this PR requires additional review of its security implications for other reasons?
      * [ ] Yes
      * [X] No

### What Needs to Be Merged and Deployed Before this PR?

This PR cannot be either merged or deployed until the following prerequisite changes have been fully deployed:

* N/A

### Submitter Checklist

I have gone through and verified that...:

* [X] I have named this PR and branch so they are [automatically linked](https://confluence.atlassian.com/adminjiracloud/integrating-with-development-tools-776636216.html) to the (most) relevant Jira issue. Ie: `BFD-123: Adds foo`
* [X] This PR is reasonably limited in scope, to help ensure that:
    1. It doesn't unnecessarily tie a bunch of disparate features, fixes, refactorings, etc. together.
    2. There isn't too much of a burden on reviewers.
    3. Any problems it causes have a small "blast radius".
    4. It'll be easier to rollback if that becomes necessary.
* [X] This PR includes any required documentation changes, including `README` updates and changelog / release notes entries.
* [X] The data dictionary has been updated with any field mapping changes, if any were made.
* [X] All new and modified code is appropriately commented, such that the what and why of its design would be reasonably clear to engineers, preferably ones unfamiliar with the project.
* [X] All tech debt and/or shortcomings introduced by this PR are detailed in `TODO` and/or `FIXME` comments, which include a JIRA ticket ID for any items that require urgent attention.
* [X] Reviews are requested from both:
    * At least two other engineers on this project, at least one of whom is a senior engineer or owns the relevant component(s) here.
    * Any relevant engineers on other projects (e.g. DC GEO, BB2, etc.).
* [X] Any deviations from the other policies in the [DASG Engineering Standards](https://github.com/CMSgov/cms-oeda-dasg/blob/master/policies/engineering_standards.md) are specifically called out in this PR, above.
    * Please review the standards every few months to ensure you're familiar with them.

### CODE REVIEW

File: apps/bfd-server/bfd-server-war/src/main/java/gov/cms/bfd/server/war/r4/providers/TransformerUtilsV2.java [hunk 1]
```diff
@@ -3766,17 +3766,23 @@ static Observation findOrCreateContainedObservation(ExplanationOfBenefit eob, St
    * @param id The resource ID
    * @return The found or new {@link Organization} resource
    */
-  static Resource findOrCreateContainedOrg(ExplanationOfBenefit eob, String id) {
-    Optional<Resource> org = eob.getContained().stream().filter(r -> r.getId() == id).findFirst();
+  static Organization findOrCreateContainedOrganization(ExplanationOfBenefit eob, String id) {
+    Optional<Resource> organization =
+        eob.getContained().stream().filter(r -> r.getId() == id).findFirst();
```
Comment (lsmitchell) [inline apps/bfd-server/bfd-server-war/src/main/java/gov/cms/bfd/server/war/r4/providers/TransformerUtilsV2.java:3771]: Are these non unit test related changes meant to be in this PR?

File: apps/bfd-server/bfd-server-war/src/test/java/gov/cms/bfd/server/war/r4/providers/TransformerUtilsV2Test.java [hunk 2]
```diff
@@ -271,6 +274,49 @@ public void mapEobCommonGroupInpOutHHAHospiceSNFWhenNoFiNumberExpectNoFiNumExten
...
+        BigDecimal.ZERO,
+        Optional.empty(),
+        Optional.of(instant),
+        Optional.empty(),
+        Optional.empty(),
+        C4BBInstutionalClaimSubtypes.Inpatient,
+        Optional.empty());
+
+    assertNotNull(eob.getContained());
+    Organization actualEobContainedOrganizationResource = (Organization) eob.getContained().get(0);
```
Comment (lsmitchell) [inline apps/bfd-server/bfd-server-war/src/test/java/gov/cms/bfd/server/war/r4/providers/TransformerUtilsV2Test.java:304]: Might be worth a getContained size assertion to make sure its 1 (duplicate mapping check), also avoids an exception if it the container to contained exists but has no elements.

File: apps/bfd-server/bfd-server-war/src/test/java/gov/cms/bfd/server/war/r4/providers/TransformerUtilsV2Test.java [hunk 3]
```diff
@@ -872,4 +918,117 @@ public void createBundleWithoutPaging() throws IOException {
     assertEquals(1, bundle.getTotal());
     assertEquals(1, Integer.parseInt(BfdMDC.get("resources_returned_count")));
   }
+
+  /**
+   * Verifies that {@link TransformerUtilsV2#findOrCreateContainedOrganization} sets organization
+   * and eob correctly.
+   */
+  @Test
+  public void findOrCreateContainedOrganizationSetsOrganizationAndEobCorrectly()
```
Comment (lsmitchell) [inline apps/bfd-server/bfd-server-war/src/test/java/gov/cms/bfd/server/war/r4/providers/TransformerUtilsV2Test.java:927]: Is this testing the finding or creating part of the method? (Ideally we'd test both cases)

File: apps/bfd-server/bfd-server-war/src/main/java/gov/cms/bfd/server/war/r4/providers/TransformerUtilsV2.java (hunk 1 above)
Comment (cbrunefearless) [reply apps/bfd-server/bfd-server-war/src/main/java/gov/cms/bfd/server/war/r4/providers/TransformerUtilsV2.java:3771]: yes

Comment (cbrunefearless) [reply apps/bfd-server/bfd-server-war/src/main/java/gov/cms/bfd/server/war/r4/providers/TransformerUtilsV2.java:3771]: i wanted to refactor some of the functions that gets hit.. Just made more sense to me

File: apps/bfd-server/bfd-server-war/src/test/java/gov/cms/bfd/server/war/r4/providers/TransformerUtilsV2Test.java (hunk 3 above)
Comment (cbrunefearless) [reply apps/bfd-server/bfd-server-war/src/test/java/gov/cms/bfd/server/war/r4/providers/TransformerUtilsV2Test.java]: Added a unit test for find and a unit test for create

**Review by lsmitchell — DISMISSED** [review]
Thanks for the adjustments!

**Review by lsmitchell — APPROVED** [review]

**Review by mjburling — APPROVED** [review]

**Pull request merged**